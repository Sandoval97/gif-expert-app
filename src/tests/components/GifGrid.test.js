import { shallow } from "enzyme"
import GifGrid from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs"

jest.mock('../../hooks/useFetchGifs')

describe('Pruebas al componente GifGrid', () => {
    let category = 'Match'

    test('Debe mostrar el componente correctamente', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: false
        })
        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper).toMatchSnapshot()
    })

    test('Se debe de renderizar el texto de carga porque no hay items', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow(<GifGrid category={category} />)
        expect(wrapper.find('p').exists()).toBe(true)
    })

    test('Se debe de renderizar el numero correcto de items', () => {

        let gifs = [{
            id: 0,
            url: 'https://localhost/test.gif',
            title: 'Imagen Prueba'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        const wrapper = shallow(<GifGrid category={category} />)

        expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
    })
})