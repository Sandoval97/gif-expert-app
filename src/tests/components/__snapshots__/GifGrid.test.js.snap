// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas al componente GifGrid Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Match
  </h3>
  <div
    className="card-grid"
  />
</Fragment>
`;
